src/js/game/Entities/BaseEntity.js

```diff
@@ -212,7 +213,12 @@ module.exports = class BaseEntity {
         let forwardPosition = this.controller.levelModel.getMoveForwardPosition(this);
         var forwardPositionInformation = this.controller.levelModel.canMoveForward(this);
         if (forwardPositionInformation[0]) {
+            let offset = this.controller.directionToOffset(this.facing);
+            let weMovedOnTo = this.handleMoveOnPressurePlate(offset);
             this.doMoveForward(commandQueueItem, forwardPosition);
+            if (!weMovedOnTo) {
+              this.handleMoveOffPressurePlate(this.reverseOffset(offset));
+            }
```

Comment:
Looks great 🎉

The only case I see this might not handle is multiple entities both standing on the pressure plate, then one steps off.  But that's such an edge case I don't think we're worried about it for now.  This is a big improvement!

Reply:
I actually think that's handled elsewhere (unless I botched something here).